mock-identity-certify-plugin/src/main/java/io.mosip.certify.mock.integration/service/MockVCIssuancePlugin.java

```diff
@@ -0,0 +1,127 @@
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.mosip.certify.api.exception.VCIExchangeException;
+import io.mosip.certify.api.util.ErrorConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import foundation.identity.jsonld.ConfigurableDocumentLoader;
+import foundation.identity.jsonld.JsonLDException;
+import foundation.identity.jsonld.JsonLDObject;
+import info.weboftrust.ldsignatures.LdProof;
+import info.weboftrust.ldsignatures.canonicalizer.URDNA2015Canonicalizer;
+import io.mosip.certify.api.dto.VCRequestDto;
+import io.mosip.certify.api.dto.VCResult;
+import io.mosip.certify.api.spi.VCIssuancePlugin;
+import io.mosip.kernel.core.util.CryptoUtil;
+import io.mosip.kernel.signature.dto.JWTSignatureRequestDto;
+import io.mosip.kernel.signature.dto.JWTSignatureResponseDto;
+import io.mosip.kernel.signature.service.SignatureService;
+import lombok.extern.slf4j.Slf4j;
+
+@ConditionalOnProperty(value = "mosip.certify.integration.vci-plugin", havingValue = "MockVCIssuancePlugin")
+@Component
+@Slf4j
+public class MockVCIssuancePlugin implements VCIssuancePlugin {
+	@Autowired
+	private SignatureService signatureService;
+
+	private ConfigurableDocumentLoader confDocumentLoader = null;
+
+	@Value("${mosip.certify.mock.vciplugin.verification-method}")
+	private String verificationMethod;
+
+	public static final String UTC_DATETIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+
+	public static final String OIDC_SERVICE_APP_ID = "CERTIFY_SERVICE";
+
+	@Override
+	public VCResult<JsonLDObject> getVerifiableCredentialWithLinkedDataProof(VCRequestDto vcRequestDto, String holderId,
+			Map<String, Object> identityDetails) throws VCIExchangeException {
+		JsonLDObject vcJsonLdObject = null;
+		try {
+			VCResult vcResult = new VCResult();
+			vcJsonLdObject = buildDummyJsonLDWithLDProof(holderId);
+			vcResult.setCredential(vcJsonLdObject);
+			vcResult.setFormat("ldp_vc");
+			return vcResult;
+		} catch (Exception e) {
+			log.error("Failed to build mock VC", e);
+		}
+		throw new VCIExchangeException();
+	}
+
+	private JsonLDObject buildDummyJsonLDWithLDProof(String holderId)
+			throws IOException, GeneralSecurityException, JsonLDException, URISyntaxException {
+		Map<String, Object> formattedMap = new HashMap<>();
+		formattedMap.put("id", holderId);
+		formattedMap.put("name", "John Doe");
+		formattedMap.put("email", "john.doe@mail.com");
+		formattedMap.put("gender", "Male");
```

Comment:
this will be hardcoded ? Can we not take it from a list of config ? I believe in eSignet we were integration to mock id system

Reply:
took it from existing esignet-mock-services [code](https://github.com/mosip/esignet-mock-services/blob/develop/mock-esignet-integration-impl/src/main/java/io/mosip/esignet/mock/integration/service/MockVCIssuancePlugin.java#L172).

Reply:
But they make call to mock id system and fill the data right ?

Reply:
changed to take from api call